sqlglot/expressions.py

```diff
@@ -201,6 +201,21 @@ def find_ancestor(self, *expression_types):
             ancestor = ancestor.parent
         return ancestor
 
+    def find_ancestor_with_parent(self, *expression_types):
```

Comment:
maybe this hould just be a parameter on find_ancestor?

Reply:
I prefer the separate function but open to combining the two if there a strong vote for that.